_posts/2018-04-01-from-chemical-engineering-to-software-development.md

```diff
@@ -0,0 +1,114 @@
+
+In my previous life:
+- all engineers have access to a shared disk
+- you can not edit a file (or it is complicated) if someone has already opened it
+- you pick a file `my-file-v32.txt`, copy it, make some enhancements, rename the copy to `my-file-v33.txt`, store `v32` in a folder called `old` with the 31 other files
+- unless you have studied to Hogwarts, it is not possible to clearly see all the changes between `v32` and `v33`
+- you can easily mess up in the shared folder and lose 4 hours of your time with the documentation controller
+
+And no, all of this did not happen in the 80s, I have been undergoing a bunch of documentation crises when I started to work in the chemical industry in 2015...
+
+As software engineers, we are using a version control software, the most famous one is [`git`](https://git-scm.com/):
+- all engineers have access to a shared remote repository which has a "master" state
+- if you want to alter a part of the code, you can create another version from the initial version of the code ("branching"), everybody can do so, there is no interference
+- once you are happy with your enhancements, there is a review process ("pull request") - every single changed line is visible
+- if successfully approved, your version becomes the master version ("merging), available for the other engineers
+
+## Diversity of careers and domains
+
+As I started as a chemical engineer, two main destinies stood out:
+- do some repetitive and low-level tasks as a beginner for few years (4, 5...), then luckily start managing other juniors and become a senior, then maybe lead a team, and after decades maybe become chief of a department (if you are lucky enough: this kinds of position are rare and people live long nowadays)
+- still do some boring junior tasks for some years, then start to dig in a very specific area of expertise (e.g. the pumps), then become an expert referent and do that for the rest of your career (yes, designing complicated pumps for 50 years).
+
+Of course, I only describe (maybe with a little exaggeration) the paths for a chemical engineer in an engineering office context, chemical engineer on production fields (e.g. offshore platforms) is something very different.
+
+Things are shaped differently in the software industry. As long as you can write a line of code, you can become: software developer, data scientist/analyst, automation/manual tester, DevOps engineer, embedded software engineer, software architect, implementation engineer...
+Furthermore, upgrading your role level is pretty fair and is based on your skills, rather than being based on non-scientific parameters like the number of years spent with the role of "chief manager of a team composed of me and myself", or how grey my hair is (or no hair at all). If someone has all the required potential to reach the Moon, just let her/him go.
```

Comment:
'upgrading your role level is pretty fair' - I would probably add 'usually' in here. I think we're both lucky with the companies we've encountered, and no doubt there are still plenty of software-based companies that don't always hire and promote based on these factors.

Reply:
Arg, you've got a damn fair point here!